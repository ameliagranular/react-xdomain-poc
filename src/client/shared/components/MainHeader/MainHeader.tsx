/**
 * The main header of the application.
 */

import * as React from 'react';
import './MainHeader.scss';

import { MainNav } from './components/MainNav';

const MainHeader: React.StatelessComponent = () => (
    <div className="MainHeader">
        <MainNav/>
    </div>
);

export { MainHeader };
