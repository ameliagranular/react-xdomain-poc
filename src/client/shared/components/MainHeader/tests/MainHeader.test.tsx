import 'jest'; // necessary for types

import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';

import { MainHeader } from '../MainHeader';

// Configure Enzyme for use with Jest & ES6+
configure({ adapter: new Adapter() });


const wrapper = (props = {}) => (
    <MainHeader {...props} />
);

const doMount = (props = {}) => {
    return mount(wrapper(props));
};

describe('MainHeader', () => {
    it('renders expected markup', () => {
        expect(toJson(doMount())).toMatchSnapshot();
    });
});
