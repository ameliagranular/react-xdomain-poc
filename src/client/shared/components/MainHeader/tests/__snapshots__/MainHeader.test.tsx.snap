// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders expected markup 1`] = `
<MainHeader>
  <div
    className="MainHeader"
  >
    <MainNav>
      <div
        className="MainNav text-center"
      >
        <div
          className="MainNav_Title"
        >
          Granular
        </div>
      </div>
    </MainNav>
  </div>
</MainHeader>
`;
